app/services/geocode_api_client.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'net/http'
+require 'uri'
+require 'json'
+
+# get latitude and longitude for city
+class GeocodeApiClient
+  def initialize(city:, state:)
+    @city = city
+    @state = state
+    @base_url = ENV['GEOCODE_API_BASE_URL']
```

Comment:
keys should be in secrets, i believe. Accessing them from there is more secure and better practice